client/src/components/UserPassword.js

```diff
@@ -1,39 +1,55 @@
 // @flow
 
 import type { EditContainerProps } from '@performant-software/shared-components/types';
-import React from 'react';
+import React, { useMemo } from 'react';
 import { useTranslation } from 'react-i18next';
 import { Form, Message } from 'semantic-ui-react';
 
 type Props = EditContainerProps;
 
+const SSO_DOMAINS = process.env.REACT_APP_SSO_DOMAINS
+  ? process.env.REACT_APP_SSO_DOMAINS.split(',')
+  : [];
+
 const UserPassword = (props: Props) => {
   const { t } = useTranslation();
 
-  return (
-    <>
-      <Message
-        content={t('UserPassword.messages.password.content')}
-        header={t('UserPassword.messages.password.header')}
-      />
-      <Form.Input
-        error={props.isError('password')}
-        label={t('UserPassword.labels.password')}
-        onChange={props.onTextInputChange.bind(this, 'password')}
-        required={props.isRequired('password')}
-        type='password'
-        value={props.item.password || ''}
-      />
-      <Form.Input
-        error={props.isError('password_confirmation')}
-        label={t('UserPassword.labels.passwordConfirmation')}
-        onChange={props.onTextInputChange.bind(this, 'password_confirmation')}
-        required={props.isRequired('password_confirmation')}
-        type='password'
-        value={props.item.password_confirmation || ''}
-      />
-    </>
-  );
+  const showPasswordFields = useMemo(() => {
+    if (props.item.email || props.item.user?.email) {
```

Comment:
It looks like we may be making some assumptions about the data type of the `item` prop here. Would it be beneficial for the parent component to make the determination on whether to include the `UserPassword` component? We could extract the `SSO_DOMAINS` checking logic to a utils function.